dev/TEAL.md

```diff
@@ -44,19 +44,19 @@ Starting from version 2 TEAL evaluator can run programs in two modes:
 2. Application run (stateful)
 
 Differences between modes include:
-1. Max program length (consensus parameters LogicSigMaxSize, MaxApprovalProgramLen and MaxClearStateProgramLen)
+1. Max program length (consensus parameters LogicSigMaxSize, MaxAppProgramLen & MaxExtraAppProgramPages)
```

Comment:
Yes, and no.  The real bounds are derived from MaxAppTotalProgramLen (which is one "page" - it's called "Total" to reflect that it is the sum of Approval and ClearState) and MaxExtraAppProgramPages (which is the number of such program lengths).  So the actual test is:
```
		lap := len(tx.ApprovalProgram)
		lcs := len(tx.ClearStateProgram)
		pages := int(1 + tx.ExtraProgramPages)
		if lap+lcs > pages*proto.MaxAppTotalProgramLen {
			return fmt.Errorf("app programs too long. max total len %d bytes", pages*proto.MaxAppTotalProgramLen)
		}
```
There are also individual tests against MaxAppProgramLen, but those are essentially no-ops in the new protocol level because MaxAppProgramLen  and MaxAppTotalProgramLen are now the same.

I'll make the correction.